pkg/resource/statefulset.go

```diff
@@ -157,14 +157,27 @@ func (b StatefulSetBuilder) makePodTemplate() corev1.PodTemplateSpec {
 			AutomountServiceAccountToken:  ptr.Bool(false),
 		},
 	}
+
+	secret := b.Spec().Image.PullSecret
+	if secret != nil {
+		local := corev1.LocalObjectReference{
+			Name: *secret,
+		}
+
+		var imagePullSecrets []corev1.LocalObjectReference
+		imagePullSecrets[0] = local
+		pod.Spec.ImagePullSecrets = imagePullSecrets
```

Comment:
won't this result in a nil reference as slices default to nil?

Would this be considered more idiomatic?
```suggestion
		pod.Spec.ImagePullSecrets = []corev1.LocalObjectReference{local}
```

Reply:
Yes, and I can never remember the damn syntax 😂